gl.cpp

```diff
@@ -348,30 +396,35 @@ COLOR colorRGB(const GLFix r, const GLFix g, const GLFix b)
 
 GLFix nglZBufferAt(const unsigned int x, const unsigned int y)
 {
-    if(x >= SCREEN_WIDTH || y >= SCREEN_HEIGHT)
+    if (x >= SCREEN_WIDTH || y >= SCREEN_HEIGHT)
         return 0;
 
     return z_buffer[x + y * SCREEN_WIDTH];
 }
 
 //Doesn't interpolate colors even if enabled
-void nglDrawLine3D(const VERTEX *v1, const VERTEX *v2)
-{    
+void nglDrawLine3D(const VERTEX* v1, const VERTEX* v2)
+{
     //TODO: Z-Clipping
 
     VERTEX v1_p = *v1, v2_p = *v2;
     nglPerspective(&v1_p);
     nglPerspective(&v2_p);
 
     const GLFix diff_x = v2_p.x - v1_p.x;
+
+#ifndef _TINSPIRE
+    const GLFix dy = diff_x != GLFix(0) ? (v2_p.y - v1_p.y) / diff_x : (v2_p.y - v1_p.y);
```

Comment:
Yep, this should work!

It could even be simplified/optimized a bit: this value is initially used for deciding whether this line should be drawn from bottom to top (first case) or left to right (second case). It could be replaced by comparing the absolute `diff_x` with the absolute `diff_y` and only calculating `dy` when actually used in the second case.